src/cli/node/json_rpc_client.rs

```diff
@@ -345,6 +348,288 @@ pub fn master_key_export(
+    #[serde(skip)]
+    print_data_request: bool,
+}
+
+#[derive(Debug, Serialize)]
+struct DataRequestState {
+    stage: String,
+    current_commit_round: u16,
+    current_reveal_round: u16,
+}
+
+impl fmt::Display for DataRequestTransactionInfo {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        writeln!(
+            f,
+            "Report for data request {}:",
+            White.bold().paint(&self.data_request_tx_hash)
+        )?;
+
+        if self.print_data_request {
+            writeln!(
+                f,
+                "data_request_output: {}",
+                serde_json::to_string_pretty(&self.data_request_output).unwrap()
+            )?;
+        }
+
+        if self.block_hash_data_request_tx == "pending" {
+            writeln!(
+                f,
+                "Deployed by {}, not yet included in any block",
+                self.data_request_creator_pkh
+            )?;
+        } else {
+            writeln!(
+                f,
+                "Deployed in block {} by {}",
+                Purple.bold().paint(&self.block_hash_data_request_tx),
+                self.data_request_creator_pkh
+            )?;
+            let data_request_state = self.data_request_state.as_ref().unwrap();
+            let num_commits = self.reveals.as_ref().unwrap().len();
+            let num_reveals = self
+                .reveals
+                .as_ref()
+                .unwrap()
+                .iter()
+                .filter_map(
+                    |(_pkh, reveal, _honest)| {
+                        if reveal.is_empty() {
+                            None
+                        } else {
+                            Some(())
+                        }
+                    },
+                )
+                .count();
+            if data_request_state.stage == "FINISHED" {
+                writeln!(
+                    f,
+                    "{} with {} commits and {} reveals",
+                    White.bold().paint(&data_request_state.stage),
+                    num_commits,
+                    num_reveals,
+                )?;
+            } else {
+                writeln!(
+                    f,
+                    "In {} stage with {} commits and {} reveals",
+                    White.bold().paint(&data_request_state.stage),
+                    num_commits,
+                    num_reveals,
+                )?;
+            }
+            writeln!(
+                f,
+                "Commit rounds: {}/{}",
+                data_request_state.current_commit_round,
+                1 + self.data_request_output.extra_commit_rounds
+            )?;
+            writeln!(
+                f,
+                "Reveal rounds: {}/{}",
+                data_request_state.current_reveal_round,
+                1 + self.data_request_output.extra_reveal_rounds
+            )?;
+        }
+
+        if let Some(reveals) = &self.reveals {
+            writeln!(f, "Reveals:")?;
```

Comment:
This header is printed even before we are in the `REVEAL` stage. I assumed it's because `&self.reveals` could be `Some([])` at that point.